From ed9f09b785d524586be9d405c7ad5527345e2cfd Mon Sep 17 00:00:00 2001
From: Robin Lee <rgl@google.com>
Date: Fri, 10 Sep 2021 12:03:42 +0000
Subject: [PATCH] DO NOT MERGE Initialize DrawingState::trustedOverlay to false
 in constructor

To avoid it being initialised to true randomly.

Bug: 199483370
Bug: 196389741
Change-Id: I75be2b1d305e22f8a71532b9f5b8ea6c469baaaa
Merged-In: I75be2b1d305e22f8a71532b9f5b8ea6c469baaaa
(cherry picked from commit 41f48c7b10132c94822ab109de99978ee65cc743)
Merged-In: I75be2b1d305e22f8a71532b9f5b8ea6c469baaaa
---
 services/surfaceflinger/Layer.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/services/surfaceflinger/Layer.cpp b/services/surfaceflinger/Layer.cpp
index ceea79f98..4e3ef2468 100644
--- a/services/surfaceflinger/Layer.cpp
+++ b/services/surfaceflinger/Layer.cpp
@@ -118,6 +118,7 @@ Layer::Layer(const LayerCreationArgs& args)
     mCurrentState.shadowRadius = 0.f;
     mCurrentState.treeHasFrameRateVote = false;
     mCurrentState.fixedTransformHint = ui::Transform::ROT_INVALID;
+    mCurrentState.isTrustedOverlay = false;
 
     if (args.flags & ISurfaceComposerClient::eNoColorFill) {
         // Set an invalid color so there is no color fill.
-- 
2.40.0.rc1.284.g88254d51c5-goog

